server/cli/src/cli.rs

```diff
@@ -639,3 +696,24 @@ pub struct TestReportArguments {
     pub reference_data: Option<String>,
     pub data_id: Option<String>,
 }
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+pub struct ReportData {
+    pub id: String,
+    pub name: String,
+    pub r#type: repository::ReportType,
+    pub template: ReportDefinition,
+    pub context: ContextType,
+    pub sub_context: Option<String>,
+    pub argument_schema_id: Option<String>,
```

Comment:
There is a bunch of nested structs here.
I wasn't sure the best way to do this, and doing this way adds a bit of bloat in structs.

I also didn't like how it wasn't actually type safe because fields are optional. ie the 'arguments' field in manifest.json is optional as well as the fields within arguments.
This means it is quite easy to miss spell a field resulting in it not being found and added to the large json file (and is quite easy to miss!). I wonder if there is a way to automate this? Make a test which checks the json fields? Or perhaps structuring it in a different way? having all fields but explicitly calling None / null in the fields where they are empty?

Later I will add more tests to hopefully make